src/files/infra/repositories/FilesRepository.ts

```diff
@@ -0,0 +1,92 @@
+import { ApiRepository } from '../../../core/infra/repositories/ApiRepository';
+import { IFilesRepository } from '../../domain/repositories/IFilesRepository';
+import { FileOrderCriteria } from '../../domain/models/FileOrderCriteria';
+import { File } from '../../domain/models/File';
+import { transformFilesResponseToFiles } from './transformers/fileTransformers';
+import { FileDataTable } from '../../domain/models/FileDataTable';
+import { transformDataTablesResponseToDataTables } from './transformers/fileDataTableTransformers';
+import { FileUserPermissions } from '../../domain/models/FileUserPermissions';
+import { transformFileUserPermissionsResponseToFileUserPermissions } from './transformers/fileUserPermissionsTransformers';
+
+export interface GetFilesQueryParams {
+  limit?: number;
+  offset?: number;
+  orderCriteria?: string;
+}
+
+export class FilesRepository extends ApiRepository implements IFilesRepository {
+  public async getDatasetFiles(
+    datasetId: number | string,
+    datasetVersionId?: string,
+    limit?: number,
+    offset?: number,
+    orderCriteria?: FileOrderCriteria,
+  ): Promise<File[]> {
+    if (datasetVersionId === undefined) {
+      datasetVersionId = this.DATASET_VERSION_LATEST;
+    }
```

Comment:
I think you can avoid this if by giving a default value to datasetVersionId arg
```suggestion
  public async getDatasetFiles(
    datasetId: number | string,
    datasetVersionId: string = this.DATASET_VERSION_LATEST,
    limit?: number,
    offset?: number,
    orderCriteria?: FileOrderCriteria,
  ): Promise<File[]> {
```

Reply:
@GPortas you are refactoring this in another PR right? I'm closing the thread